### PR TITLE
Update release notes for BC4

### PR DESCRIPTION
**Pull request by craigtaverner**
Regenerated release notes for BC4, and *ten* new PRs were added. See generated docs at https://elasticsearch_87635.docs-preview.app.elstc.co/diff

### CODE REVIEW

File: docs/reference/release-notes/highlights.asciidoc [hunk 1]
```diff
@@ -29,6 +29,42 @@ for compliance or regulatory reasons, the occasional lookback or investigation,
...
 and can therefore happen with limited performance and query capabilities.
 
+[discrete]
+[[new_geo_grid_query]]
+=== New geo_grid query
+With the geo grid query, you can now natively return all the documents that overlap a specific geo tile.
+There is no need to reconstruct the geometry or the actual boundaries of the spatial cluster as Elasticsearch
+can do this for you which saves you time and reduces complexity.
+This is especially useful when geometries are spread across tiles like on a soccer ball football.
+While hexagon tiles line the sphere each tile is not exactly straight forward to calculate its boundary.
```
Comment (kilfoyle) [inline docs/reference/release-notes/highlights.asciidoc:39]:
```suggestion
There is no need to reconstruct the geometry or the actual boundaries of the spatial cluster as Elasticsearch
can do this for you, which saves you time and reduces complexity.
This is especially useful when geometries are spread across tiles like on a soccer ball or football.
While hexagon tiles line the sphere, calculating the boundary of each tile is not straightforward.
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 2]
```diff
@@ -29,6 +29,42 @@ for compliance or regulatory reasons, the occasional lookback or investigation,
...
+As an example, if a client has bounds for a grid-cell at a higher (or lower) precision than what is used
+by Elasticsearch when running a corresponding aggregation, the containment-check might be slightly different.
+This side-steps any disconnect based on projection/datum difference between client and Elasticsearch.
+
+[discrete]
+[[ingest_performance_improvement]]
+=== Ingest performance improvement
+We have improved the pipeline execution logic for pipelines with processors that are synchronous
+by avoiding (deep) recursion.
+On our nightly benchmark that simulate a Logging use-case, this resulted in a 10% reduction of
```
Comment (kilfoyle) [inline docs/reference/release-notes/highlights.asciidoc:65]:
```suggestion
On our nightly benchmark that simulates a Logging use-case, this resulted in a 10% reduction of
```

**Review by kilfoyle — APPROVED** [review]
LGTM! 
Though I'd suggest waiting for a confirmation from @debadair since I'm not yet familiar with the Elasticsearch release notes process. :)

File: docs/reference/release-notes/highlights.asciidoc (hunk 1 above)
Comment (craigtaverner) [reply docs/reference/release-notes/highlights.asciidoc]: The text came from PM from a planned blog, but I think your improvements are great, so I just added them. We'll see if PM want to improve the blog later.

**Pull request merged**